api/apps/v1alpha1/nemo_guardrails_types.go

```diff
@@ -187,6 +187,17 @@ func (n *NemoGuardrail) GetStandardEnv() []corev1.EnvVar {
 			Name:  "DEMO",
 			Value: "False",
 		},
+		{
```

Comment:
the `authSecret` is optional and used only when NIMs are hosted by NGC. This should be added conditionally as well.

Reply:
`authSecret` is currently a required field for GuardRail 
https://github.com/NVIDIA/k8s-nim-operator/blob/311fc567c6d6262bcdebee9832c5bd44d9911074/config/crd/bases/apps.nvidia.com_nemoguardrails.yaml#L2207

We should make it optional if it's not always required

Reply:
`.spec.authSecret` in NemoGuardrail is now optional